test/domainRegistry.test.ts

```diff
@@ -0,0 +1,145 @@
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+// TEST contract ./contracts/domainRegistry.sol
+describe("DomainRegistry", function () {
+  let DomainRegistry: any;
+  let domainRegistry: any;
+  let addr1: SignerWithAddress;
+  let addr2: SignerWithAddress;
+
+  beforeEach(async function () {
+    [addr1, addr2] = await ethers.getSigners();
+    DomainRegistry = await ethers.getContractFactory("DomainRegistry");
+    domainRegistry = await DomainRegistry.deploy();
+  });
+
+  it("Should top level domain special characters", async function () {
+    const invalidDomainNames = [
+      ".com",
+      "com.",
+      "domain@name",
+      "special*char",
+      "example.com",
+    ];
+
+    const depositAmount = ethers.parseEther("0.001");
+    for (const domainName of invalidDomainNames) {
+      await expect(
+        domainRegistry
+          .connect(addr1)
+          .registerDomain(domainName, { value: depositAmount })
+      ).to.be.revertedWith("It is prohibited to use special characters");
+    }
+  });
+
+  it("Should with upercase", async function () {
+    const invalidDomainNames = ["UA", "Com", "eXample"];
+
+    const depositAmount = ethers.parseEther("0.001");
+    for (const domainName of invalidDomainNames) {
+      await expect(
+        domainRegistry
+          .connect(addr1)
+          .registerDomain(domainName, { value: depositAmount })
+      ).to.be.revertedWith("Domain name must be in lower case");
+    }
+  });
+
+  it("Should top level domain special characters", async function () {
+    const invalidDomainNames = [".com", "com.", "domain@name", "special*char"];
+
+    const depositAmount = ethers.parseEther("0.001");
+    for (const domainName of invalidDomainNames) {
+      await expect(
+        domainRegistry
+          .connect(addr1)
+          .registerDomain(domainName, { value: depositAmount })
+      ).to.be.revertedWith("It is prohibited to use special characters");
+    }
+  });
+
+  it("Should register a domain with a deposit", async function () {
+    const domainName = "example";
+    const depositAmount = ethers.parseEther("0.001");
+    await domainRegistry
+      .connect(addr1)
+      .registerDomain(domainName, { value: depositAmount });
+
+    const domain = await domainRegistry.domains(domainName);
+    expect(domain.owner).to.equal(await addr1.getAddress());
+    expect(domain.deposit).to.equal(depositAmount);
+  });
+
+  it("Should release a domain and refund the deposit", async function () {
+    const domainName = "example";
+    const depositAmount = ethers.parseEther("0.001");
+
+    await domainRegistry
+      .connect(addr1)
+      .registerDomain(domainName, { value: depositAmount });
+    const balanceBefore = await ethers.provider.getBalance(addr1);
+    const tx = await domainRegistry.connect(addr1).releaseDomain(domainName);
+    const receipt = await tx.wait(); // Очікуємо на підтвердження транзакції
```

Comment:
There is no reason to overload the code with comments when the code is obvious. -1 point